### PR TITLE
Add support for traits in RBAddClassRefactoring

### PR DESCRIPTION
**Pull request by sonibla**
Hello there

I added
RBAddClassRefactoring >> addClass:superclass:subclasses: **traits:** category:
to support traits in refactorings

Please correct me if I did it wrong

Old methods are still available (RBAddClassRefactoring >> addClass:superclass:subclasses:category:)

### CODE REVIEW

**Comment by MarcusDenker** [comment]
tests are failing, for mac:

```
osx-64 / Tests-osx-64 / testAddClass(#rbClass->RBAddClassRefactoring) – MacOSX64.Refactoring2.Transformations.Tests.Parametrized.RBAddClassParametrizedTest
1s
osx-64 / Tests-osx-64 / testModelAddClass(#rbClass->RBAddClassRefactoring) – MacOSX64.Refactoring2.Transformations.Tests.Parametrized.RBAddClassParametrizedTest
1s
osx-64 / Tests-osx-64 / testModelChildrenToSibling(#rbClass->RBChildrenToSiblingsRefactoring) – MacOSX64.Refactoring2.Transformations.Tests.Parametrized.RBChildrenToSiblingsParametrizedTest
2s
osx-64 / Tests-osx-64 / testCopyPackage(#rbClass->RBCopyPackageRefactoring) – MacOSX64.Refactoring2.Transformations.Tests.Parametrized.RBCopyPackageParametrizedTest
1s
osx-64 / Tests-osx-64 / testCopyPackageAndChangesCopyReferences(#rbClass->RBCopyPackageRefactoring) – MacOSX64.Refactoring2.Transformations.Tests.Parametrized.RBCopyPackageParametrizedTest
1s
osx-64 / Tests-osx-64 / testSplitClass(#rbClass->RBSplitClassRefactoring) – MacOSX64.Refactoring2.Transformations.Tests.Parametrized.RBSplitClassParametrizedTest
```

**Comment by sonibla** [comment]
I forgot the case where there are no traits being used, my bad

File: src/Refactoring-Core/RBAddClassRefactoring.class.st [hunk 1]
```diff
@@ -39,31 +51,61 @@ RBAddClassRefactoring class >> model: aRBSmalltalk addClass: aName superclass: a
...
+		              self classObjectFor: each ].
+
+	traits := #(  ).
+	category := aSymbol
+]
+
+{ #category : #initialization }
+RBAddClassRefactoring >> addClass: aName superclass: aClass subclasses: subclassesCollection traits: traitsCollection category: aSymbol [
+
+	self className: aName.
```
Comment (demarey) [inline src/Refactoring-Core/RBAddClassRefactoring.class.st]: there is code duplication here.
You should call `self addClass: aName superclass: aClass subclasses: aCollection category: aSymbol` and then only initialize traits i.v.

File: src/Refactoring-Core/RBAddClassRefactoring.class.st [hunk 2]
```diff
@@ -77,17 +119,28 @@ RBAddClassRefactoring >> storeOn: aStream [
...
+		defineClass:
+			('<1p> subclass: #<2s><3s> instanceVariableNames: '''' classVariableNames: '''' poolDictionaries: '''' category: <4p>'
+				 expandMacrosWith: superclass
+				 with: className
+				 with: (String streamContents: [ :stream | 
+							  traits ifNotEmpty: [ 
+									  stream nextPutAll: ' uses: '.
+									  traits
+										  do: [ :each | stream nextPutAll: each name ]
+										  separatedBy: [ stream nextPutAll: ' + ' ] ] ])
```
Comment (demarey) [inline src/Refactoring-Core/RBAddClassRefactoring.class.st]: probably this code handles only simple case, use a full traits but do not handle traits composition.

**Review by demarey — CHANGES_REQUESTED** [review]
Thank you for the PR.
It will be too short for Pharo10, better to target Pharo 11.
This PR is missing tests for traits support.
Also, not related to this PR, but Refactorings should be rewritten to do not use text but a model instead (like Ring).

**Comment by sonibla** [comment]
Thanks for your feedback @demarey 
I improved my PR a little bit but I don't feel it's enough, ideally RBClass and RBTrait would support trait compositions so that it's possible to test traits in RBAddClassParametrizedTest...

**Pull request closed without merging**